### DISCUSSION TITLE
Successfully authenticated via ssh, but times out when cloning

### DISCUSSION THREAD

**Discussion by NotBadForAWednesday** [community/community#22045] [New to GitHub]
<p>I’ve created an ssh key (without a passphase) by following <a href="https://docs.github.com/en/authentication/connecting-to-github-with-ssh/checking-for-existing-ssh-keys" class="inline-onebox" rel="noopener nofollow ugc">Checking for existing SSH keys - GitHub Docs</a> instructions for Mac’s. I have Git installed locally.</p>
<p>I can then authenticate.</p>
<pre><code class="lang-auto">% ssh -T git@github.com
</code></pre>
<blockquote>
<p>Hi <em>MyGitHubUsername</em>! You’ve successfully authenticated, but GitHub does not provide shell access.</p>
</blockquote>
<p>Opening a new Terminal on an empty folder locally, I want to clone my GitHub repo into it:</p>
<pre><code class="lang-auto">git clone git@github.com:*MyGitHubUsername*/*MyGitHubRepoName*.git MyRepo
</code></pre>
<blockquote>
<p>Cloning into ‘MyRepo’…<br>
ssh: connect to host 3.93.192.127 port 22: Operation timed out<br>
fatal: Could not read from remote repository.</p>
<p>Please make sure you have the correct access rights<br>
and the repository exists.</p>
</blockquote>
<p>I’ve removed all the ssh key files and entries from my .ssh folder and ssh-agent respectively and re-installed them to no avail. Why is it timing out when I can authenticate, please? How can I resolve?</p>
<p>My .ssh config file</p>
<pre><code class="lang-auto">Host github.com
  AddKeysToAgent yes
  IdentityFile ~/.ssh/id_ed25519
</code></pre>
<p>My .ssh known_host file</p>
<pre><code class="lang-auto">github.com ssh-ed25519 AAAAC3Nz_REMOVED
github.com ssh-rsa AAAAB3Nz_REMOVED+PXYPCPy6rbTrTtw7_REMOVED==
github.com ecdsa-sha2-nistp256 AAAAE2VjZHNhLXN_REMOVED=
</code></pre>

**Comment by NotBadForAWednesday** [comment, marked as answer]
<p>Self resolved.</p>
<p>The source of the problem was within my local ~/.gitconfig</p>
<p>It contained a URL for a legacy repo that no longer existed as a [remote “origin”] value which was redirecting. Removed the entry, then all worked as expected.</p>

**Reply by DanielMachado-443** [reply]
Hey man, can you show how did you set up you ~/.gitconfig file and WHERE it has to be? I'm having the same problem while trying to make  a clone of my repository from my remoted accessed EC2 instance of Linux Ubuntu server.

**Reply by nasrqwertuyy** [reply]
حل ذاتي.

كان مصدر المشكلة داخل بلدي المحلي ~ / .gitconfig

احتوت على عنوان URL لـ repo قديم لم يعد موجودًا كقيمة [أصل "بعيد"] والذي كان يعيد التوجيه. تمت إزالة الإدخال ، ثم عمل كل شيء كما هو متوقع.

**Reply by DanielMachado-443** [reply]
That's the matter, I didn't have or found this gitconfig file on my actual setup
